strategies/iCARL/experiment.py

```diff
@@ -0,0 +1,191 @@
+        np.random.seed(config.seed)
+        random.seed(config.seed)
+        torch.backends.cudnn.enabled = False
+        torch.backends.cudnn.deterministic = True
+
+        per_pixel_mean = get_dataset_per_pixel_mean(
+            CIFAR100(expanduser("~") + "/.avalanche/data/cifar100/",
+                     train=True, download=True,
+                     transform=transforms.Compose([transforms.ToTensor()])))
+
+        transforms_group = dict(
+            eval=(transforms.Compose([
+                transforms.ToTensor(),
+                lambda img_pattern: img_pattern - per_pixel_mean]), None),
+            train=(transforms.Compose([
+                transforms.ToTensor(),
+                lambda img_pattern: img_pattern - per_pixel_mean,
+                icarl_cifar100_augment_data]), None),
+        )
+
+        train_set = CIFAR100(expanduser("~") + "/.avalanche/data/cifar100/",
+                             train=True, download=True, )
+        test_set = CIFAR100(expanduser("~") + "/.avalanche/data/cifar100/",
+                            train=False, download=True, )
+
+        train_set = AvalancheDataset(train_set,
+                                     transform_groups=transforms_group,
+                                     initial_transform_group='train')
+        test_set = AvalancheDataset(test_set,
+                                    transform_groups=transforms_group,
+                                    initial_transform_group='eval')
+
+        scenario = nc_benchmark(
+            train_dataset=train_set,
+            test_dataset=test_set,
+            n_experiences=config.nb_exp,
+            task_labels=False, seed=config.seed,
+            shuffle=False,
+            fixed_class_order=config.fixed_class_order)
+
+        evaluator = EvaluationPlugin(ExperienceAccuracy(),
+                                     StreamAccuracy(),
+                                     loggers=[InteractiveLogger()])
+
+        model: IcarlNet = make_icarl_net(num_classes=100)
+        model.apply(initialize_icarl_net)
+
+        optim = SGD(model.parameters(), lr=config.lr_base,
+                    weight_decay=config.wght_decay, momentum=0.9)
+        sched = LRSchedulerPlugin(
+            MultiStepLR(optim, config.lr_milestones, gamma=1.0 / config.lr_factor))
+
+        strategy = ICaRL(
+            model.feature_extractor, model.classifier, optim,
+            config.memory_size,
+            buffer_transform=transforms.Compose([icarl_cifar100_augment_data]),
+            fixed_memory=True, train_mb_size=config.batch_size,
+            train_epochs=config.epochs, eval_mb_size=config.batch_size,
+            plugins=[sched], device=device, evaluator=evaluator
+        )
+
+        # Dict to iCaRL Evaluation Protocol: Average Incremental Accuracy
+        dict_iCaRL_aia = {}
+        # Training and eval
+        for i, exp in enumerate(scenario.train_stream):
+            eval_exps = [e for e in scenario.test_stream][:i + 1]
```

Comment:
you can call `eval` directly on the sliced test stream `strategy.eval(scenario.test_stream[:i+1])`.

Reply:
ok if `strategy.eval(scenario.test_stream[:i+1])` is equivalent to `eval_exps = [e for e in scenario.test_stream][:i + 1]` I understand :)